aws/cloudformation/components/ami.yml.erb

```
@@ -146,63 +147,86 @@ frontend_device_name ||= '/dev/sda1'
         - Key: "environment"
           Value: <%=environment%>
           PropagateAtLaunch: true
-  FrontendLaunchConfig:
-    Type: AWS::AutoScaling::LaunchConfiguration
+  FrontendLaunchTemplate:
+    Type: AWS::EC2::LaunchTemplate
     DependsOn: FastSnapshotRestore
     Properties:
-      ImageId: !GetAtt [AMI<%=ami%>, ImageId]
-      InstanceType: <%=instance_type%>
-      IamInstanceProfile: !Ref FrontendInstanceProfile
-      SecurityGroups: [!ImportValue VPC-FrontendSecurityGroup]
-      KeyName: <%=ssh_key_name%>
-      BlockDeviceMappings:
-        - DeviceName: <%=frontend_device_name%>
-          Ebs:
-            VolumeSize: <%=frontend_volume_size * 2%> # Double volume size to leave room for logs written to disk
-            VolumeType: gp2
-      UserData:
-        Fn::Base64: !Sub |
-          #!/bin/bash -x
-          # UserData bootstrap script for CloudFormation stack instances.
-          # Note: Every time this script's contents change,
-          # the LaunchConfiguration will be replaced in the CloudFormation stack.
+      LaunchTemplateName: !Sub ${AWS::StackName}-frontend
+      VersionDescription: !Sub "AMI: ${AMI<%=commit_hash%>.ImageId}, Commit: <%=commit_hash%>"
+      TagSpecifications:
+        - ResourceType: launch-template
+          Tags:
+            - Key: "git-commit"
+              Value: <%=commit_hash%>
+            - Key: "ami-id"
+              Value: !GetAtt AMI<%=commit_hash%>.ImageId
+            - Key: "environment"
+              Value: <%=environment%>
+      LaunchTemplateData:
+        ImageId: !GetAtt [AMI<%=commit_hash%>, ImageId]
+        InstanceType: <%=instance_type%>
+        IamInstanceProfile:
+          Name: !Ref FrontendInstanceProfile
+        SecurityGroupIds:
+          - !ImportValue VPC-FrontendSecurityGroup
+        KeyName: <%=ssh_key_name%>
+        BlockDeviceMappings:
+          - DeviceName: <%=frontend_device_name%>
+            Ebs:
+              VolumeSize: <%=frontend_volume_size * 2%>
+              VolumeType: gp2
+        TagSpecifications:
+          - ResourceType: instance
+            Tags:
+              - Key: "Name"
+                Value: !Sub "Frontend-${AWS::StackName}"
+              - Key: "git-commit"
+                Value: <%=commit_hash%>
+              - Key: "environment"
+                Value: <%=environment%>
+        UserData:
+          Fn::Base64: !Sub |
+            #!/bin/bash -x
+            # UserData bootstrap script for CloudFormation stack instances.
+            # Note: Every time this script's contents change,
+            # the LaunchTemplate version will need to be updated.
```

```suggestion
            # the LaunchTemplate version will be updated.
```

IMO "need to" implies that the engineer might need to do something, whereas what we're trying to say is that it will happen automatically as a side effect that the engineer should be aware of.

---

I think we can remove the comments. We don’t change this UserData script often (note that it does NOT contain the git commit hash of the current release).